### ISSUE TITLE
Loading a calculation on an existing resource does not loads its dependent loads

### ISSUE THREAD

**Issue by ahey** [ash-project/ash_postgres#127]
**Describe the bug**
When `Api.load!` is called on an existing resource to load a calculation on that resource, and that calculation itself has dependent loads specified, those dependent loads are not automatically loaded.

**To Reproduce**
Please see [this test case that reproduces the issue](https://github.com/ahey/ash_postgres/commit/df3b4fd82ce358a555c971c0c69c40eceb84c509)

**Expected behavior**
The dependent load should be loaded automatically.

** Runtime
 - Elixir version 1.14.2
 - Erlang version 25.1.1
 - OS MacOS 13
 - Ash Postgres version 1.2.3

**Comment by zachdaniel** [comment]
So is the`active` value that depends on that load correct? If the calculated value is correct, but the dependent load is just not there on the return value, that is expected/acceptable behavior. The dependent loads are used to communicate the data that is needed for a given calculation to be successful, but it's not guaranteed that those values will be loaded in the response. Additionally, those dependencies are inferred automatically for expression calculations. Anything you want to be present on the results must be specifically loaded.

If, however, the `active` field is incorrect or something, that's a different story.

Thanks for the report!

**Comment by ahey** [comment]
Hi @zachdaniel in this case it's actually throwing an exception:

```elixir
1) test loading a calculation loads its dependent loads (AshPostgres.CalculationTest)
     test/calculation_test.exs:337
     ** (RuntimeError) Error while building reference: user_is_active
     code: |> Api.load!([:active])
     stacktrace:
       (ash_postgres 1.2.3) lib/expr.ex:576: AshPostgres.Expr.do_dynamic_expr/5
       (ash_postgres 1.2.3) lib/expr.ex:391: AshPostgres.Expr.do_dynamic_expr/5
       (ash_postgres 1.2.3) lib/calculation.ex:21: anonymous fn/2 in AshPostgres.Calculation.add_calculations/3
       (elixir 1.14.2) lib/enum.ex:1658: Enum."-map/2-lists^map/1-0-"/2
       (ash_postgres 1.2.3) lib/calculation.ex:19: AshPostgres.Calculation.add_calculations/3
       (ash 2.4.25) lib/ash/actions/read.ex:1770: Ash.Actions.Read.run_calculation_query/4
       (ash 2.4.25) lib/ash/actions/read.ex:1562: anonymous fn/6 in Ash.Actions.Read.calculation_request/11
       (ash 2.4.25) lib/ash/engine/engine.ex:460: anonymous fn/2 in Ash.Engine.run_iteration/1
       (ash 2.4.25) lib/ash/engine/engine.ex:615: Ash.Engine.start_pending_tasks/1
       (ash 2.4.25) lib/ash/engine/engine.ex:265: Ash.Engine.run_to_completion/1
       (ash 2.4.25) lib/ash/engine/engine.ex:194: Ash.Engine.do_run/2
       (ash 2.4.25) lib/ash/engine/engine.ex:133: Ash.Engine.run/2
       (ash 2.4.25) lib/ash/actions/read.ex:170: Ash.Actions.Read.do_run/3
       (ash 2.4.25) lib/ash/actions/read.ex:90: Ash.Actions.Read.run/3
       (ash 2.4.25) lib/ash/api/api.ex:810: Ash.Api.load/4
       (ash 2.4.25) lib/ash/api/api.ex:784: Ash.Api.load!/4
       test/calculation_test.exs:354: (test)
```

**Comment by zachdaniel** [comment]
Alright, I see the issue :) AshPostgres is supposed to know to add the aggregates it depends on the hood, but it wasn't doing that. For other readers, its not expected that it will actually load the dependent values (i.e they may or may not be selected into the response), but it is expected that you can refer to an aggregate from a calculation and have the ash_postgres data layer know what to do.

**Comment by zachdaniel** [comment]
Fix released in 1.2.4

**Comment by ahey** [comment]
Thank you for this @zachdaniel, it's working beautifully.
